harness/determined/pytorch/lightning/_adapter.py

```diff
@@ -266,18 +277,31 @@ def lightning_scheduler_dict_to_det(lrs: dict) -> LRScheduler:
                 if lrs["interval"] == "epoch"
                 else LRScheduler.StepMode.STEP_EVERY_BATCH
             )
-            opt = getattr(lrs["scheduler"], "optimizer", None)
-            if opt is not None:
-                check.is_in(
-                    opt,
-                    self._pls.optimizers,
-                    "Must use an optimizer that is returned by wrap_optimizer()",
+
+            wrapped_opt = optimizers_dict[getattr(lrs["scheduler"], "optimizer", None)]
+            if wrapped_opt is None:
+                raise InvalidModelException(
+                    "An LRScheduler is returned in `configure_optimizers` without having "
+                    "returned the optimizer itself. Please follow PyTorchLightning's documenation"
+                    "to make sure you're returning one of the expected values."
+                    "- Single optimizer.\n"
+                    "- List or Tuple - List of optimizers.\n"
+                    "- Two lists - The first list has multiple optimizers, the second a list of"
+                    "LRSchedulers (or lr_dict).\n"
+                    "- Dictionary, with an ‘optimizer’ key, and (optionally) a ‘lr_scheduler’ key"
+                    "whose value is a single LR scheduler or lr_dict.\n"
+                    "- Tuple of dictionaries as described, with an optional ‘frequency’ key.\n"
                 )
-            return LRScheduler(lrs["scheduler"], step_mode, frequency=lrs["frequency"])
 
-        optimizers = [self._pls.context.wrap_optimizer(opt) for opt in optimizers]
+            # switch the user's unwrapped optimizer with the wrapped version.
+            lrs["scheduler"].optimizer = wrapped_opt
```

Comment:
Can you also emit a check here that the scheduler object actually has an `optimizer` attribute?

Sorry, I didn't understand the check above, I thought that is what I was asking for before.

But the error message above looks fantastic : )

Reply:
: ) . it seems like the base lrscheduler has a [similar check ](https://github.com/pytorch/pytorch/blob/master/torch/optim/lr_scheduler.py#L28) but doesn't hurt to have it here as well.

Reply:
Yes, but the issue is that people pass LRSchedulers which are not actually API compatible with the real _LRScheduler at all